runtime-watcher/internal/cacertificatehandler/ca_certificate_handler_test.go

```diff
@@ -0,0 +1,131 @@
+package cacertificatehandler_test
+
+import (
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/kyma-project/runtime-watcher/skr/internal/cacertificatehandler"
+	"github.com/kyma-project/runtime-watcher/skr/internal/tlstest"
+)
+
+func TestGetCertificatePool1(t *testing.T) {
+	t.Parallel()
+	certPath := "ca.cert"
+	tests := []struct {
+		name             string
+		certificateCount int
+	}{
+		{
+			name:             "certificate pool with one certificate",
+			certificateCount: 1,
+		},
+		{
+			name:             "certificate pool with two certificates",
+			certificateCount: 2,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
We are executing these test cases in parallel but it's a single file being written. A race condition may occur.